assets/src/edit-story/components/form/dateTime/datePicker.js

```diff
@@ -1,101 +1,49 @@
-import 'react-dates/lib/css/_datepicker.css';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
 import { useRef, useCallback } from 'react';
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { rgba } from 'polished';
 
-/**
- * Internal dependencies
- */
-import { useConfig } from '../../../app';
-import { TIMEZONELESS_FORMAT } from '../../../constants';
-
 const CalendarWrapper = styled.div`
   min-height: 236px;
   border-top: 1px solid ${({ theme }) => rgba(theme.colors.bg.v0, 0.2)};
 `;
 
 function DatePicker({ currentDate, onChange }) {
-  const { isRTL } = useConfig();
-  const momentDate = currentDate ? moment(currentDate) : moment();
-
   const nodeRef = useRef();
-  const onChangeMoment = useCallback(
+  const value = new Date(currentDate);
+  const handleOnChange = useCallback(
     (newDate) => {
-      const momentTime = {
-        hours: momentDate.hours(),
-        minutes: momentDate.minutes(),
-        seconds: 0,
-      };
-      onChange(
-        newDate.set(momentTime).format(TIMEZONELESS_FORMAT),
-        /* Close calendar */ true
-      );
+      newDate.setHours(value.getHours());
+      newDate.setMinutes(value.getMinutes());
+      onChange(newDate.toISOString(), /* Close calendar */ true);
     },
-    [momentDate, onChange]
+    [value, onChange]
   );
 
-  /**
-   * Note: This comment is copied from Gutenberg.
-   *
-   * It is kept because focus is lost when we click on the previous and next month buttons.
-   * This focus loss closes the date picker popover.
-   * Ideally we should add an upstream commit on react-dates to fix this issue.
-   */
-  const keepFocusInside = () => {
-    if (!nodeRef.current) {
-      return;
-    }
-    // If focus was lost.
-    if (
-      !document.activeElement ||
-      !nodeRef.current.contains(document.activeElement)
-    ) {
-      // Retrieve the focus region div.
-      const focusRegion = nodeRef.current.querySelector(
-        '.DayPicker_focusRegion'
-      );
-      if (!focusRegion) {
-        return;
-      }
-      // Keep the focus on focus region.
-      focusRegion.focus();
-    }
-  };
-
   return (
     <CalendarWrapper ref={nodeRef}>
-      <DayPickerSingleDateController
-        date={momentDate}
-        daySize={30}
-        focused
-        hideKeyboardShortcutsPanel
-        // This is a hack to force the calendar to update on month or year change
-        // https://github.com/airbnb/react-dates/issues/240#issuecomment-361776665
-        key={`datepicker-controller-${
-          momentDate ? momentDate.format('MM-YYYY') : 'null'
-        }`}
-        noBorder
-        numberOfMonths={1}
-        onDateChange={onChangeMoment}
-        transitionDuration={0}
-        weekDayFormat="ddd"
-        isRTL={isRTL}
-        onPrevMonthClick={keepFocusInside}
-        onNextMonthClick={keepFocusInside}
-      />
+      <Calendar value={value} onChange={handleOnChange} />
```

Comment:
This new component can change size - e.g. when you switch from month view to year view. And that thas the unfortunate consequence of potentially popping outside the viewport without the sidebar correcting for it, as the sidebar doesn't listen for potential size changes of the currently shown element.

This is not directly a bug from this PR, but still a weird situation (note that I have the web console open in the bottom to show that it's the bottom of the browser window here):

![calendar-resize-scroll](https://user-images.githubusercontent.com/637548/82392952-21c77f80-9a13-11ea-8a36-9e6324439072.gif)

Reply:
@barklund "Fixed" this by allowing setting the max height to the popup with scrolling. The popup seems to have CSS for scrolling implemented, however, it didn't really seem to support scrolling.

That's something that needs looking into within the popup logic and is out of scope here indeed. Perhaps the popup should have some kind of default max height (and width) anyway and allow changing it. (cc @wassgha )

I can leave as-is with the scrolling fix or revert, leave the bug in, and create a separate issue to address it generally in the popup.

Thoughts?

Reply:
Let's file an issue for this! Valid feature request for the popup

Reply:
Added: https://github.com/google/web-stories-wp/issues/1931

Reply:
Fine with me!